ssl/statem/statem_clnt.c

```diff
@@ -3280,6 +3280,125 @@ static int tls_construct_cke_gost(SSL *s, WPACKET *pkt)
 #endif
 }
 
+static int tls_construct_cke_gost18(SSL *s, WPACKET *pkt)
+{
+#ifndef OPENSSL_NO_GOST
+    /* GOST 2018 key exchange message creation */
+    unsigned char rnd_dgst[32], tmp[255];
+    unsigned int md_len;
+    EVP_MD_CTX * hash = NULL;
+    EVP_PKEY_CTX *pkey_ctx = NULL;
+    X509 *peer_cert;
+    unsigned char *pms = NULL;
+    size_t pmslen = 0;
+    int cipher_nid = NID_undef;
+    size_t msglen;
+
+    if ((s->s3.tmp.new_cipher->algorithm_enc & SSL_MAGMA) != 0)
+        cipher_nid = NID_magma_ctr;
+    else if ((s->s3.tmp.new_cipher->algorithm_enc & SSL_KUZNYECHIK) != 0)
+        cipher_nid = NID_grasshopper_ctr;
+    else {
```

Comment:
i think when the else is using {} all other branches should do that as well

Reply:
Fixed.